integritybackend/actions.py

```diff
@@ -658,235 +673,100 @@ def copy_proofmode(self, zip_path: str, org_config: dict, collection_id: str):
+            zip_path: path to asset zip (will be copied, not altered)
+            org_config: configuration dictionary for this organization
+            collection_id: string with the unique collection identifier this asset is in
+
+        Returns:
+            the local path to the asset file in the internal directory
+
+        Raises:
+            Exception if errors are encountered during processing
+        """
+
+        # Setup context
+
+        action_name = "c2pa-starling-capture"
+        org_id = org_config["id"]
+
+        asset_helper = AssetHelper(org_id)
+
+        collection = config.ORGANIZATION_CONFIG.get_collection(org_id, collection_id)
+        action = config.ORGANIZATION_CONFIG.get_action(
+            org_id, collection_id, action_name
+        )
+        action_params = action.get("params")
+
+        content_filename, content_sha = self._verify_zip(
+            zip_path, collection["asset_extensions"]
+        )
+
+        # Extract content file
+        tmp_dir = asset_helper.path_for_action_tmp(collection_id, action_name)
+        zip_dir = os.path.join(tmp_dir, content_sha)
+        extracted_content = os.path.join(zip_dir, content_filename)
+        _file_util.create_dir(zip_dir)
+        zip_util.extract_file(zip_path, content_filename, extracted_content)
+
+        # Load meta content
+        meta_content_filename = f"{content_sha}-meta-content.json"
+        meta_content = zip_util.json_load(zip_path, meta_content_filename)
+
+        # Create temporary files to work with.
+        tmp_asset_file = asset_helper.get_tmp_file_fullpath(".jpg")
+        tmp_claim_file = asset_helper.get_tmp_file_fullpath(".json")
+
+        # Inject create claim and read back from file.
+        claim = _claim.generate_c2pa_starling_capture(meta_content["contentMetadata"])
+        shutil.copy2(extracted_content, tmp_asset_file)
+        _c2patool.run_claim_inject(
+            claim,
+            tmp_asset_file,
+            tmp_asset_file,
+            action_params["c2pa_cert"],
+            action_params["c2pa_key"],
+            action_params["c2pa_algo"],
+        )
+        _c2patool.run_claim_dump(tmp_asset_file, tmp_claim_file)
+
+        # Copy the C2PA-injected asset to both the internal and shared asset directories.
+        asset_file_hash = _file_util.digest_sha256(tmp_asset_file)
+        internal_asset_file = os.path.join(
+            asset_helper.path_for_action(collection_id, action_name),
+            "assets",
+            asset_file_hash + ".jpg",
+        )
+        shutil.move(tmp_asset_file, internal_asset_file)
+        shutil.copy2(
+            internal_asset_file,
+            os.path.join(
+                asset_helper.path_for_action_output(collection_id, action_name),
+                meta_content["author"].get("name", "unknown"),
+                datetime.datetime.now(timezone.utc).strftime("%Y-%m-%d"),
+            ),
+        )
+        # Keep manifest in internal dir but not in shared dir
```

Comment:
Is the placement of JPG and manifest dumps consistent with `c2pa_proofmode`?

Reply:
The images are the same, but in `c2pa_proofmode`, the manifest dump actually never leaves the tmp folder and gets deleted in the end. Is that the behaviour we want to copy, or should that be changed in `c2pa_proofmode`?

Reply:
Change `c2pa_proofmode` to save the dump.

Reply:
Upon further investigation, it seems that in `c2pa_starling_capture`, the manifest dump ends up in the internal action directory, but not the shared directory. In `c2pa_proofmode`, it turns out the manifests end up in both. What behaviour should we settle on?

Reply:
>the manifest dump ends up in the internal action directory

Yes that is what I remember of the behaviour.

We should handle it the same as the c2pa asset itself. If the c2pa asset is only in shared, then we do the same with the manifest json. If the c2pa asset is in both, then the manifest json should be in both.

Reply:
I have updated `c2pa_starling_capture` so that the manifest is also copied to the shared dir along with the asset. This should resolve this discussion and bring the behaviour in-line across all actions.